### PR TITLE
ON-7453 # Added project setup

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    "sass-loader": "^9.0.3",
+    "script-loader": "^0.7.2",
+    "signature_pad": "^3.0.0-beta.3",
+    "style-loader": "^1.2.1",
+    "validate.js": "^0.13.1",
+    "webpack": "^4.44.1",
+    "webpack-cli": "^3.3.12",
+    "webpack-dev-server": "^3.11.0",
+    "webpack-merge": "^5.1.1",
+    "whatwg-fetch": "^3.4.0"
```
Comment (mymattcarroll) [inline package.json:84]: This should not be here IMO, should be up to the consumer to include polyfills or not

File: package.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    "sanitize-html": "^1.27.3",
+    "sass-loader": "^9.0.3",
+    "script-loader": "^0.7.2",
+    "signature_pad": "^3.0.0-beta.3",
+    "style-loader": "^1.2.1",
+    "validate.js": "^0.13.1",
+    "webpack": "^4.44.1",
+    "webpack-cli": "^3.3.12",
+    "webpack-dev-server": "^3.11.0",
+    "webpack-merge": "^5.1.1",
```
Comment (mymattcarroll) [inline package.json:83]: Can remove all this webpack stuff

File: package.json [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    "query-string": "^6.13.1",
+    "react-google-recaptcha": "^2.1.0",
+    "react-router-dom": "^5.2.0",
+    "react-signature-canvas": "^1.0.3",
+    "regenerator-runtime": "^0.13.7",
+    "sanitize-html": "^1.27.3",
+    "sass-loader": "^9.0.3",
+    "script-loader": "^0.7.2",
+    "signature_pad": "^3.0.0-beta.3",
+    "style-loader": "^1.2.1",
```
Comment (mymattcarroll) [inline package.json:78]: Remove

File: package.json [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    "phonegap-plugin-barcodescanner": "^8.1.0",
+    "quagga": "^0.12.1",
+    "query-string": "^6.13.1",
+    "react-google-recaptcha": "^2.1.0",
+    "react-router-dom": "^5.2.0",
+    "react-signature-canvas": "^1.0.3",
+    "regenerator-runtime": "^0.13.7",
+    "sanitize-html": "^1.27.3",
+    "sass-loader": "^9.0.3",
+    "script-loader": "^0.7.2",
```
Comment (mymattcarroll) [inline package.json:76]: Remove

File: package.json [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    "morph-expressions": "^1.1.1",
+    "node-sass": "^4.14.1",
+    "offline-plugin": "^5.0.7",
+    "phonegap-plugin-barcodescanner": "^8.1.0",
+    "quagga": "^0.12.1",
+    "query-string": "^6.13.1",
+    "react-google-recaptcha": "^2.1.0",
+    "react-router-dom": "^5.2.0",
+    "react-signature-canvas": "^1.0.3",
+    "regenerator-runtime": "^0.13.7",
```
Comment (mymattcarroll) [inline package.json:73]: Remove

File: package.json [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    "jsqr": "^1.3.1",
+    "jwt-decode": "^2.2.0",
+    "local-storage": "^2.0.0",
+    "lodash.clonedeep": "^4.5.0",
+    "lodash.debounce": "^4.0.8",
+    "lodash.throttle": "^4.1.1",
+    "morph-expressions": "^1.1.1",
+    "node-sass": "^4.14.1",
+    "offline-plugin": "^5.0.7",
+    "phonegap-plugin-barcodescanner": "^8.1.0",
```
Comment (mymattcarroll) [inline package.json:67]: i think make the cordova stuff either peer or not here at all and document that it should be included if intending to create a native app. Im leaning towards to the latter, prob worth opening up to the wide team.

File: package.json [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    "html-webpack-plugin": "^4.3.0",
+    "jsqr": "^1.3.1",
+    "jwt-decode": "^2.2.0",
+    "local-storage": "^2.0.0",
+    "lodash.clonedeep": "^4.5.0",
+    "lodash.debounce": "^4.0.8",
+    "lodash.throttle": "^4.1.1",
+    "morph-expressions": "^1.1.1",
+    "node-sass": "^4.14.1",
+    "offline-plugin": "^5.0.7",
```
Comment (mymattcarroll) [inline package.json:66]: Remove

File: package.json [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    "html-loader": "^1.1.0",
+    "html-webpack-plugin": "^4.3.0",
+    "jsqr": "^1.3.1",
+    "jwt-decode": "^2.2.0",
+    "local-storage": "^2.0.0",
+    "lodash.clonedeep": "^4.5.0",
+    "lodash.debounce": "^4.0.8",
+    "lodash.throttle": "^4.1.1",
+    "morph-expressions": "^1.1.1",
+    "node-sass": "^4.14.1",
```
Comment (mymattcarroll) [inline package.json:65]: Remove

File: package.json [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    "escape-string-regexp": "^4.0.0",
+    "file-loader": "^6.0.0",
+    "file-saver": "^2.0.2",
+    "flatpickr": "4.6.3",
+    "getusermedia": "^2.0.1",
+    "html-loader": "^1.1.0",
+    "html-webpack-plugin": "^4.3.0",
+    "jsqr": "^1.3.1",
+    "jwt-decode": "^2.2.0",
+    "local-storage": "^2.0.0",
```
Comment (mymattcarroll) [inline package.json:60]: I dont think this is being used in our Form

File: package.json [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    "element-remove": "^1.0.4",
+    "escape-string-regexp": "^4.0.0",
+    "file-loader": "^6.0.0",
+    "file-saver": "^2.0.2",
+    "flatpickr": "4.6.3",
+    "getusermedia": "^2.0.1",
+    "html-loader": "^1.1.0",
+    "html-webpack-plugin": "^4.3.0",
+    "jsqr": "^1.3.1",
+    "jwt-decode": "^2.2.0",
```
Comment (mymattcarroll) [inline package.json:59]: Remove

File: package.json [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    "css-loader": "^4.2.1",
+    "dotenv": "^8.2.0",
+    "element-remove": "^1.0.4",
+    "escape-string-regexp": "^4.0.0",
+    "file-loader": "^6.0.0",
+    "file-saver": "^2.0.2",
+    "flatpickr": "4.6.3",
+    "getusermedia": "^2.0.1",
+    "html-loader": "^1.1.0",
+    "html-webpack-plugin": "^4.3.0",
```
Comment (mymattcarroll) [inline package.json:57]: Remove

File: package.json [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+    "core-js": "^3.6.5",
+    "css-loader": "^4.2.1",
+    "dotenv": "^8.2.0",
+    "element-remove": "^1.0.4",
+    "escape-string-regexp": "^4.0.0",
+    "file-loader": "^6.0.0",
+    "file-saver": "^2.0.2",
+    "flatpickr": "4.6.3",
+    "getusermedia": "^2.0.1",
+    "html-loader": "^1.1.0",
```
Comment (mymattcarroll) [inline package.json:56]: Remove

File: package.json [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+    "cordova-plugin-whitelist": "^1.3.4",
+    "core-js": "^3.6.5",
+    "css-loader": "^4.2.1",
+    "dotenv": "^8.2.0",
+    "element-remove": "^1.0.4",
+    "escape-string-regexp": "^4.0.0",
+    "file-loader": "^6.0.0",
+    "file-saver": "^2.0.2",
+    "flatpickr": "4.6.3",
+    "getusermedia": "^2.0.1",
```
Comment (mymattcarroll) [inline package.json:55]: Remove

File: package.json [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    "cordova-plugin-network-information": "^2.0.2",
+    "cordova-plugin-splashscreen": "^6.0.0",
+    "cordova-plugin-statusbar": "^2.4.3",
+    "cordova-plugin-whitelist": "^1.3.4",
+    "core-js": "^3.6.5",
+    "css-loader": "^4.2.1",
+    "dotenv": "^8.2.0",
+    "element-remove": "^1.0.4",
+    "escape-string-regexp": "^4.0.0",
+    "file-loader": "^6.0.0",
```
Comment (mymattcarroll) [inline package.json:52]: Remove

File: package.json [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+    "cordova-plugin-file-opener2": "^3.0.4",
+    "cordova-plugin-geolocation": "^4.0.2",
+    "cordova-plugin-ionic-webview": "^5.0.0",
+    "cordova-plugin-network-information": "^2.0.2",
+    "cordova-plugin-splashscreen": "^6.0.0",
+    "cordova-plugin-statusbar": "^2.4.3",
+    "cordova-plugin-whitelist": "^1.3.4",
+    "core-js": "^3.6.5",
+    "css-loader": "^4.2.1",
+    "dotenv": "^8.2.0",
```
Comment (mymattcarroll) [inline package.json:49]: Remove

File: package.json [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+    "cordova-plugin-file": "^6.0.2",
+    "cordova-plugin-file-opener2": "^3.0.4",
+    "cordova-plugin-geolocation": "^4.0.2",
+    "cordova-plugin-ionic-webview": "^5.0.0",
+    "cordova-plugin-network-information": "^2.0.2",
+    "cordova-plugin-splashscreen": "^6.0.0",
+    "cordova-plugin-statusbar": "^2.4.3",
+    "cordova-plugin-whitelist": "^1.3.4",
+    "core-js": "^3.6.5",
+    "css-loader": "^4.2.1",
```
Comment (mymattcarroll) [inline package.json:48]: Remove

File: package.json [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+    "cordova-plugin-exit": "^1.0.3",
+    "cordova-plugin-file": "^6.0.2",
+    "cordova-plugin-file-opener2": "^3.0.4",
+    "cordova-plugin-geolocation": "^4.0.2",
+    "cordova-plugin-ionic-webview": "^5.0.0",
+    "cordova-plugin-network-information": "^2.0.2",
+    "cordova-plugin-splashscreen": "^6.0.0",
+    "cordova-plugin-statusbar": "^2.4.3",
+    "cordova-plugin-whitelist": "^1.3.4",
+    "core-js": "^3.6.5",
```
Comment (mymattcarroll) [inline package.json:47]: Remove

File: package.json [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+    "blueimp-load-image": "^5.14.0",
+    "bulma": "^0.8.2",
+    "bulma-badge": "3.0.0",
+    "bulma-slider": "^2.0.4",
+    "bulma-steps": "^2.2.1",
+    "bulma-toast": "^2.0.1",
+    "bulma-tooltip": "^3.0.2",
+    "clsx": "^1.1.1",
+    "copy-to-clipboard": "^3.3.1",
+    "copy-webpack-plugin": "^6.0.3",
```
Comment (mymattcarroll) [inline package.json:32]: Remove

File: package.json [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+    "@blinkmobile/canvas-manipulation": "^1.0.1",
+    "@blinkmobile/geolocation": "^2.0.3",
+    "@oneblink/apps": "0.1.0-beta.13",
+    "@react-google-maps/api": "^1.9.12",
+    "abort-controller": "^3.0.0",
+    "abortcontroller-polyfill": "^1.5.0",
+    "babel-loader": "^8.1.0",
+    "blueimp-load-image": "^5.14.0",
+    "bulma": "^0.8.2",
+    "bulma-badge": "3.0.0",
```
Comment (mymattcarroll) [inline package.json:25]: This is not being used by the Form, I think remove.

File: package.json [hunk 20]
```diff
@@ -0,0 +1,131 @@
...
+    "@babel/preset-flow": "^7.10.4",
+    "@babel/preset-react": "^7.10.4",
+    "@blinkmobile/camera": "^0.0.4",
+    "@blinkmobile/canvas-manipulation": "^1.0.1",
+    "@blinkmobile/geolocation": "^2.0.3",
+    "@oneblink/apps": "0.1.0-beta.13",
+    "@react-google-maps/api": "^1.9.12",
+    "abort-controller": "^3.0.0",
+    "abortcontroller-polyfill": "^1.5.0",
+    "babel-loader": "^8.1.0",
```
Comment (mymattcarroll) [inline package.json:22]: Remove

File: package.json [hunk 21]
```diff
@@ -0,0 +1,131 @@
...
+    "@babel/preset-env": "^7.11.0",
+    "@babel/preset-flow": "^7.10.4",
+    "@babel/preset-react": "^7.10.4",
+    "@blinkmobile/camera": "^0.0.4",
+    "@blinkmobile/canvas-manipulation": "^1.0.1",
+    "@blinkmobile/geolocation": "^2.0.3",
+    "@oneblink/apps": "0.1.0-beta.13",
+    "@react-google-maps/api": "^1.9.12",
+    "abort-controller": "^3.0.0",
+    "abortcontroller-polyfill": "^1.5.0",
```
Comment (mymattcarroll) [inline package.json:21]: Remove

File: package.json [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+  "author": "OneBlink <developers@oneblink.io> (https://oneblink.io)",
+  "bugs": {
+    "url": "https://github.com/oneblink/apps-react/issues"
+  },
+  "dependencies": {
+    "@babel/core": "^7.11.1",
+    "@babel/plugin-transform-flow-strip-types": "^7.10.4",
+    "@babel/preset-env": "^7.11.0",
+    "@babel/preset-flow": "^7.10.4",
+    "@babel/preset-react": "^7.10.4",
```
Comment (mymattcarroll) [inline package.json:14]: Remove

File: package.json (hunk 9 above)
Comment (Zaxist) [reply package.json:60]: in `src/job/job-list-component.js`

File: package.json [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+  "version": "0.1.0",
+  "author": "OneBlink <developers@oneblink.io> (https://oneblink.io)",
+  "bugs": {
+    "url": "https://github.com/oneblink/apps-react/issues"
+  },
+  "dependencies": {
+    "@babel/core": "^7.11.1",
+    "@babel/plugin-transform-flow-strip-types": "^7.10.4",
+    "@babel/preset-env": "^7.11.0",
+    "@babel/preset-flow": "^7.10.4",
```
Comment (mymattcarroll) [inline package.json:13]: Should be able to remove all of these babel dependencies

File: package.json [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+    "@blinkmobile/camera": "^0.0.4",
+    "@blinkmobile/canvas-manipulation": "^1.0.1",
+    "@blinkmobile/geolocation": "^2.0.3",
+    "@oneblink/apps": "0.1.0-beta.13",
```
Comment (mymattcarroll) [inline package.json:17]: I think make this a peer as well.

Comment (mymattcarroll) [reply package.json:17]: Ignore that, changed my mind 👍

**Pull request merged**